tests/st/calicoctl/test_upgrade.py

```diff
@@ -0,0 +1,624 @@
+             'nat-outgoing': False}
+}
+data['node_long_name'] = {
+    'apiVersion': 'v1',
+    'kind': 'node',
+    'metadata': {
+        'name': '-Mary_had-A-Little___Lamb--Whose---Fleece-Was-White.As.Snow...She-Also-Had_an-Evil-NodeName_in_order_to.break.upgrade-code201600'},
+    'spec': {'bgp': {'asNumber': '7.20',
+                     'ipv4Address': '10.244.0.1/24',
+                     'ipv6Address': '2001:db8:85a3::8a2e:370:7334/120'}}
+}
+data['node_tame'] = {
+    'apiVersion': 'v1',
+    'kind': 'node',
+    'metadata': {'name': 'node-hostname'},
+    'spec': {'bgp': {'asNumber': 64512,
+                     'ipv4Address': '10.244.0.1/24',
+                     'ipv6Address': '2001:db8:85a3::8a2e:370:7334/120'}}
+}
+data['policy_tame'] = {
+    'apiVersion': 'v1',
+    'kind': 'policy',
+    'metadata': {'name': 'allow-tcp-6379'},
+    'spec': {'egress': [{'action': 'allow'}],
+             'ingress': [{'action': 'allow',
+                          'destination': {'ports': [6379]},
+                          'protocol': 'tcp',
+                          'source': {'selector': "role == 'frontend'"}}],
+             'selector': "role == 'database'",
+             'types': ['ingress', 'egress']}
+}
+data['policy_long_name'] = {
+    'apiVersion': 'v1',
+    'kind': 'policy',
+    'metadata': {
+        'name': '-Mary_had-A-Little___Lamb--Whose---Fleece-Was-White.As.Snow...She-Also-Had_an-Evil-PolicyName_in_order_to.break.upgrade-code2016'},
+    'spec': {'egress': [{'action': 'allow'}],
+             'ingress': [{'action': 'allow',
+                          'destination': {'ports': [6379]},
+                          'protocol': 'tcp',
+                          'source': {'nets': ['192.168.0.1/32']}}],
+             'selector': "role == 'database'",
+             'types': ['ingress', 'egress']}
+}
+data['profile_tame'] = {
+    'apiVersion': 'v1',
+    'kind': 'profile',
+    'metadata': {'labels': {'profile': 'profile1'}, 'name': 'profile1'},
+    'spec': {'egress': [{'action': 'allow'}],
+             'ingress': [{'action': 'deny',
+                          'source': {'nets': ['10.0.20.0/24']}},
+                         {'action': 'allow',
+                          'source': {'selector': "profile == 'profile1'"}}]}
+}
+data['profile_long_labels'] = {
+    'apiVersion': 'v1',
+    'kind': 'profile',
+    'metadata': {'labels': {
+        '8roper/evil-02/key_name.which/is-also_very.long...1234567890//proper/evil-02/key_name.which/is-also_very.long...1234567890//proper/evil-02/Key_Name.which/is-also_very.long...1234567890//proper/evil-02/key_name.which/is-also_very.long...1234567890//proper/evil-02/Key_name.which/is-also_very.long...1234567890//proper/evil-02/key_name.which/is-also_very.long...1234567890//proper/evil-02/key_Name.which/is-also_very.long...1234567890//proper/evil-02/key_name.which/is-also_very.long...1234567890//proper/evil-02/9': 'frontendFrontEnd.0123456789-_-23wdffrontendFrontEnd.0124679-_-0'},
```

Comment:
can't have this one either. We use k8s label validator, so you can't have so many `/`s in the label.
See: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go#L37-L60

Reply:
too 😈 ! :)

Reply:
I don't have a problem with the failure, but the failure message needs to be clear about what the problem is so that the user can fix it.

Reply:
OK - I've left this in, but expected to FAIL with a comment explaining that failure error message isn't great, but we're not fixing it for now.